client/src/components/ClientControlPanel.tsx

```diff
@@ -82,126 +88,244 @@ const ClientControlPanel = () => {
+    return (-60 / 9) * delay + 6000 / 9;
+  };
 
   return (
     <>
       {currentPage === "Home" && (
         <div className="clientcontrolpanel__card">
           <div>
-            Grid size:{" "}
+            <div>
+              Grid size:{" "}
+              <input
+                id="mazeSize"
+                name="mazeSize"
+                type="number"
+                min="10"
+                max="50"
+                value={mazeSize}
+                onChange={(e): void => setMazeSize(Number(e.target.value))}
+              />
+            </div>
+            <div>
+              Solving Algorithm:{" "}
+              <select
+                name="algorithm"
+                id="algorithm"
+                value={algorithm}
+                onChange={(e): void => {
+                  setAlgorithm(e.target.value as keyof typeof algorithms);
+                  clearMaze();
+                }}
+              >
+                {Object.keys(algorithms).map((key) => (
+                  <option id={key} key={key} value={key}>
+                    {key}
+                  </option>
+                ))}
+              </select>
+            </div>
+            <div>
+              Generating Algorithm:{" "}
+              <select
+                name="generatingAlgorithm"
+                id="generatingAlgorithm"
+                value={generatingAlgorithm}
+                onChange={(e): void => {
+                  if (setGeneratingAlgorithm)
+                    setGeneratingAlgorithm(
+                      e.target.value as keyof typeof generateMazeAlgorithms,
+                    );
+                  clearMaze();
+                }}
+              >
+                {Object.keys(generateMazeAlgorithms).map((key) => (
+                  <option id={key} key={key} value={key}>
+                    {key}
+                  </option>
+                ))}
+              </select>
+            </div>
+            {!generating ? (
+              <button onClick={generateMaze}>Generate Maze</button>
+            ) : (
+              <>
+                <button onClick={clearMaze}>Stop</button>
+              </>
+            )}
+            {!visualize ? (
+              <button onClick={startSolving}>Visualize Solver</button>
+            ) : (
+              <>
+                <button onClick={clearMaze}>Reset</button>
+              </>
+            )}
             <input
-              id="mazeSize"
-              name="mazeSize"
-              type="number"
-              min="10"
-              max="50"
-              value={mazeSize}
-              onChange={(e): void => setMazeSize(Number(e.target.value))}
+              type="range"
+              min={10}
+              max={100}
+              step={10}
+              onChange={(e) =>
+                setDelay(delayCalculation(parseFloat(e.target.value)))
+              }
+              value={Math.round(((delay - 6000 / 9) * 9) / -60)}
```

Comment:
This math func is used in many places, make it into a util func

Reply:
moved to util module